fix build with gcc-4.3:

In file included from /usr/include/boost/date_time/local_time/tz_database.hpp:15,
                 from /usr/include/boost/date_time/local_time/local_time.hpp:20,
                 from /usr/include/boost/date_time.hpp:16,
                 from src/core/common/loghelper_server.cpp:33:
/usr/include/boost/date_time/tz_db_base.hpp:161: error: declaration of ‘typedef class boost::date_time::dst_adjustment_offsets<typename time_zone_type::time_duration_type> boost::date_time::tz_db_base<time_zone_type, rule_type>::dst_adjustment_offsets’
/usr/include/boost/date_time/time_zone_base.hpp:76: error: changes meaning of ‘dst_adjustment_offsets’ from ‘class boost::date_time::dst_adjustment_offsets<typename time_zone_type::time_duration_type>’


--- boost_1_37_0/boost/date_time/tz_db_base.hpp~	2008-02-27 21:00:24.000000000 +0100
+++ boost_1_37_0/boost/date_time/tz_db_base.hpp	2008-12-26 19:13:40.601581002 +0100
@@ -158,7 +158,7 @@
       typedef typename time_zone_type::base_type time_zone_base_type;
       typedef typename time_zone_type::time_duration_type time_duration_type;
       typedef time_zone_names_base<char_type> time_zone_names;
-      typedef dst_adjustment_offsets<time_duration_type> dst_adjustment_offsets;
+      typedef boost::date_time::dst_adjustment_offsets<time_duration_type> dst_adjustment_offsets;
       typedef std::basic_string<char_type> string_type;
 
       //! Constructs an empty database
